src/module/system/action-macros/athletics/shove.ts

```diff
@@ -50,10 +69,16 @@ function shove(options: SkillActionOptions): void {
 }
 
 class ShoveActionVariant extends SingleCheckActionVariant {
-    protected override checkContext<ItemType extends ItemPF2e<ActorPF2e>>(
-        opts: CheckContextOptions<ItemType>,
-        data: CheckContextData<ItemType>,
-    ): CheckContext<ItemType> | undefined {
+    override async use(options: ShoveActionUseOptions = {}): Promise<CheckResultCallback[]> {
+        return ActionMacroHelpers.resolveWieldedWeaponFromItemParameter(options, REQUIRED_WEAPON_TRAIT).then(
+            super.use.bind(this),
+        );
```

Comment:
There isn't really a good reason to use `then` in an async environment

Reply:
I guess awaiting should work just as well. Updated.